# Sets the correct access attributes for register lists of LDM instructions.
# See issue: https://github.com/llvm/llvm-project/issues/62455

diff --git a/arch/ARM/ARMGenCSMappingInsnOp.inc b/arch/ARM/ARMGenCSMappingInsnOp.inc
index fe71aa54f..a3b04d915 100644
--- a/arch/ARM/ARMGenCSMappingInsnOp.inc
+++ b/arch/ARM/ARMGenCSMappingInsnOp.inc
@@ -9073,7 +9073,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -9092,7 +9092,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -9108,7 +9108,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -9127,7 +9127,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -9143,7 +9143,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -9162,7 +9162,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -9178,7 +9178,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -9197,7 +9197,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -55518,7 +55518,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -55534,7 +55534,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -55553,7 +55553,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -55574,7 +55574,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -55590,7 +55590,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -55609,7 +55609,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -80998,7 +80998,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -81017,7 +81017,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -81033,7 +81033,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -81052,7 +81052,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -81068,7 +81068,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -81087,7 +81087,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -81103,7 +81103,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -81122,7 +81122,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -82831,7 +82831,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -82850,7 +82850,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -82866,7 +82866,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -82885,7 +82885,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
@@ -88945,7 +88945,7 @@
 	      CS_AC_READ,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* p - i32imm */
 	    { CS_OP_IMM,
-	      CS_AC_READ,
+	      CS_AC_WRITE,
 	      { CS_DATA_TYPE_i32, CS_DATA_TYPE_LAST } }, /* regs */
 	    { CS_OP_INVALID, CS_AC_READ, { CS_DATA_TYPE_LAST } }, /*  */
 	    { 0 } }
